Travis builds are failing again

**Summary:** 

Seems like we should have a perpetual issue for this - example of failed build:
https://travis-ci.org/OneBusAway/onebusaway-android/builds/178311476

This is related to Google renaming the ABIs again - see my past SO post:
http://stackoverflow.com/questions/28751111/invalid-abi-armeabi-v7a-for-the-selected-target-with-google-apis-21

I believe based on my answer http://stackoverflow.com/a/28751112/937715, we need to update things as specified in the **"EDIT Sept 12th, 2016"** section.  This has worked for these two projects:
* https://github.com/barbeau/gpstest
* https://github.com/googlemaps/android-maps-utils

So the API Level 22 emulator should work because the ABIs exist, especially when using the `travis_wait` command:
https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received

However, with this build:
https://travis-ci.org/OneBusAway/onebusaway-android/builds/183263286

...I'm now seeing this error:

~~~
02:55:13 E/SplitApkInstaller: Failed to finalize session : INSTALL_FAILED_DEXOPT: Package couldn't be installed in /data/app/com.joulespersecond.seattlebusbot-1: scanPackageLI
Unable to install /home/travis/build/OneBusAway/onebusaway-android/onebusaway-android/build/outputs/apk/onebusaway-android-oba-google-debug-v2.1.5.apk
com.android.ddmlib.InstallException: Failed to finalize session : INSTALL_FAILED_DEXOPT: Package couldn't be installed in /data/app/com.joulespersecond.seattlebusbot-1: scanPackageLI
	at com.android.ddmlib.SplitApkInstaller.install(SplitApkInstaller.java:85)
	at com.android.ddmlib.Device.installPackages(Device.java:904)
	at com.android.builder.testing.ConnectedDevice.installPackages(ConnectedDevice.java:137)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:126)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:49)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
com.android.builder.testing.ConnectedDevice > runTests[test(AVD) - 5.1.1] FAILED 
	com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: Failed to finalize session : INSTALL_FAILED_DEXOPT: Package couldn't be installed in /data/app/com.joulespersecond.seattlebusbot-1: scanPackageLI
		at com.android.builder.testing.ConnectedDevice.installPackages(ConnectedDevice.java:147)
null
com.android.builder.testing.api.DeviceException: com.android.ddmlib.InstallException: Failed to finalize session : INSTALL_FAILED_DEXOPT: Package couldn't be installed in /data/app/com.joulespersecond.seattlebusbot-1: scanPackageLI
	at com.android.builder.testing.ConnectedDevice.installPackages(ConnectedDevice.java:147)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:126)
	at com.android.builder.internal.testing.SimpleTestCallable.call(SimpleTestCallable.java:49)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: com.android.ddmlib.InstallException: Failed to finalize session : INSTALL_FAILED_DEXOPT: Package couldn't be installed in /data/app/com.joulespersecond.seattlebusbot-1: scanPackageLI
	at com.android.ddmlib.SplitApkInstaller.install(SplitApkInstaller.java:85)
	at com.android.ddmlib.Device.installPackages(Device.java:904)
	at com.android.builder.testing.ConnectedDevice.installPackages(ConnectedDevice.java:137)
	... 8 more
:onebusaway-android:connectedObaGoogleDebugAndroidTest FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task ':onebusaway-android:connectedObaGoogleDebugAndroidTest'.
~~~

Now all builds (even not emulator based) are failing due to a license issue - from https://travis-ci.org/OneBusAway/onebusaway-android/builds/557364172:

~~~
> Configure project :onebusaway-android
WARNING: Configuration 'compile' is obsolete and has been replaced with 'implementation' and 'api'.
It will be removed at the end of 2018. For more information see: http://d.android.com/r/tools/update-dependency-configurations.html
File /home/travis/.android/repositories.cfg could not be loaded.
Checking the license for package Android SDK Platform 29 in /usr/local/android-sdk/licenses
Warning: License for package Android SDK Platform 29 not accepted.
FAILURE: Build failed with an exception.
* What went wrong:
A problem occurred configuring project ':onebusaway-android'.
> Failed to install the following Android SDK packages as some licences have not been accepted.
     platforms;android-29 Android SDK Platform 29
  To build this project, accept the SDK license agreements and install the missing components using the Android Studio SDK Manager.
  Alternatively, to transfer the license agreements from one workstation to another, see http://d.android.com/r/studio-ui/export-licenses.html
  
  Using Android SDK: /usr/local/android-sdk
* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
* Get more help at https://help.gradle.org
BUILD FAILED in 52s
The command "./gradlew assembleObaGoogleDebug assembleObaAmazonDebug assembleYrtGoogleDebug -PdisablePreDex" exited with 1.
~~~

--- a/.travis.yml
+++ b/.travis.yml
@@ -74,7 +74,7 @@ install:
   - adb shell input keyevent 82 &
 
 script:
-  - ./gradlew assembleObaGoogleDebug assembleObaAmazonDebug assembleYrtGoogleDebug
+  - ./gradlew test check connectedObaGoogleDebugAndroidTest -x lint --stacktrace
 
 before_cache:
   - rm -f $HOME/.gradle/caches/modules-2/modules-2.lock

--- a/.travis.yml
+++ b/.travis.yml
@@ -6,6 +6,8 @@ jdk:
 env:
   global:
     - ADB_INSTALL_TIMEOUT=8
+    - BUILD_TOOLS=28.0.3
+    - COMPILE_SDK_VERSION=29
     - ABI=x86_64
     - EMU_FLAVOR=default # use google_apis flavor if no default flavor emulator
     # PATH order is incredibly important. e.g. the 'emulator' script exists in more than one place!
@@ -13,39 +15,21 @@ env:
     - TOOLS=${ANDROID_HOME}/tools
     - PATH=${ANDROID_HOME}:${ANDROID_HOME}/emulator:${TOOLS}:${TOOLS}/bin:${ANDROID_HOME}/platform-tools:${PATH}
   matrix:
-    #- API=15 # only runs locally. Create+Start once from AndroidStudio to init sdcard. Then only from command-line w/-engine classic
-    #- API=16 ABI=armeabi-v7a AUDIO=-no-audio # Fails with "com.android.builder.testing.ConnectedDevice > No tests found."
-    #- API=17 ABI=armeabi-v7a # Fails with "com.android.builder.testing.ConnectedDevice > No tests found."
-    #- API=18 ABI=armeabi-v7a # Fails with "com.android.builder.testing.ConnectedDevice > No tests found."
-    #- API=19 ABI=armeabi-v7a # Fails: kernel/emulator mismatch, and emulator-29+ doesn't support "-engine classic" to fix
+    # - API=15 # ERROR: only runs locally. Create+Start once from AndroidStudio to init sdcard. Then only from command-line w/-engine classic
+    # - API=16 ABI=armeabi-v7a AUDIO=-no-audio # ERROR: Fails with "com.android.builder.testing.ConnectedDevice > No tests found."
+    # - API=17 ABI=armeabi-v7a # Emulator runs, but GradientTest.testInterpolateColor[test(AVD) - 7.1.1] fails with java.lang.AssertionError: expected:<-65434> but was:<-65433>
+    # - API=18 ABI=armeabi-v7a # Emulator runs, but GradientTest.testInterpolateColor[test(AVD) - 7.1.1] fails with java.lang.AssertionError: expected:<-65434> but was:<-65433>
+    # - API=19 ABI=armeabi-v7a # ERROR: This AVD's configuration is missing a kernel file! Please ensure the file "kernel-ranchu" is in the same location as your system image.
     # API 20 was Android Wear only
-    - API=21
-    - API=22
-    - API=23
-    - API=24
-    - API=25
-    #- API=26 ABI=x86_64 # Fails with unrecognized tests? orchestrator change or something?
-    #- API=27 # Same failure as below for API28, but less frequently
-    #- API=28 # Fails on adb 2 second getprop timeout and get API getprop takes 2.2s, so emulator boots but adb won't install on unknown API
-    #- API=Q  # Fails at some point on emulator startup or detecting startup complete
-
-jobs:
-  fast_finish: true  # We can report success without waiting for jobs that we
-  # allow / expect to fail.
-  include:
-    - jdk: oraclejdk11
-      env: API=24
-    # The test stage is implicit, we only need to define a post-test stage for codacy finalization
-    - stage: finalize_coverage
-      env: FINALIZE_COVERAGE=TRUE API=NONE
-      install: skip
-      script: echo finalize codacy coverage uploads
-  allow_failures:
-    - jdk: oraclejdk11
-    # API=28/Q are probably fixable but we don't need allow_failures to be on
-    # in master to work through them.
-    #- env: API=28
-    #- env: API=Q
+    # - API=21 # testSetOccupancyVisibilityAndColor fails with NoSuchMethodError: No virtual method getColor()
+    # - API=22 # testSetOccupancyVisibilityAndColor fails with NoSuchMethodError: No virtual method getColor()
+    # - API=23 # testSetOccupancyVisibilityAndColor fails with NoSuchMethodError: No virtual method getColor()
+    # - API=24 # Works
+    # - API=25 # Works
+    # - API=26 ABI=x86_64 # Works
+    # - API=27 ABI=x86_64 # Works
+    - API=28 ABI=x86_64 # Works
+    # - API=29 # ERROR: Emulator startup stalls, ends with "emulator: got message from guest system fingerprint HAL"
 
 android:
   components:
@@ -57,14 +41,13 @@ licenses:
   - 'android-sdk-license-.+'
   - 'google-gdk-license-.+'
 
-# Emulator Management: Create, Start and Wait
 install:
   - echo 'count=0' > /home/travis/.android/repositories.cfg # Avoid harmless sdkmanager warning
   - echo y | sdkmanager "platform-tools" >/dev/null
   - echo y | sdkmanager "tools" >/dev/null # A second time per Travis docs, gets latest versions
-  - echo y | sdkmanager "build-tools;28.0.3" >/dev/null # Implicit gradle dependency - gradle drives changes
+  - echo y | sdkmanager "build-tools;$BUILD_TOOLS" >/dev/null # Implicit gradle dependency - gradle drives changes
   - echo y | sdkmanager "platforms;android-$API" >/dev/null # We need the API of the emulator we will run
-  - echo y | sdkmanager "platforms;android-28" >/dev/null # We need the API of the current compileSdkVersion from gradle.properties
+  - echo y | sdkmanager "platforms;android-$COMPILE_SDK_VERSION" >/dev/null # We need the API of the current compileSdkVersion from gradle.properties
   - echo y | sdkmanager --channel=4 "emulator" >/dev/null # Use canary channel to get emulator 29.x for x86_64 image support
   - echo y | sdkmanager "extras;android;m2repository" >/dev/null
   - echo y | sdkmanager "system-images;android-$API;$EMU_FLAVOR;$ABI" >/dev/null # install our emulator

--- a/onebusaway-android/build.gradle
+++ b/onebusaway-android/build.gradle
@@ -59,6 +59,9 @@ android {
         multiDexEnabled true
 
         testInstrumentationRunner "androidx.test.runner.AndroidJUnitRunner"
+
+        // This enables us to tell when we're running unit tests on Travis (#1010)
+        buildConfigField("String", "TRAVIS", "\"" + System.getenv('TRAVIS') + "\"")
     }
 
     /**

--- a/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
+++ b/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
@@ -20,6 +20,7 @@ import org.junit.Assume;
 import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
+import org.onebusaway.android.BuildConfig;
 import org.onebusaway.android.io.test.ObaTestCase;
 import org.onebusaway.android.mock.Resources;
 import org.onebusaway.android.nav.NavigationServiceProvider;
@@ -59,10 +60,9 @@ public class NavigationTest extends ObaTestCase {
     @BeforeClass
     public static void beforeClass() {
         // Only run this test suite if we're not running on Travis CI (it's too slow)
-        String runningOnTravis = System.getenv("RUNNING_ON_TRAVIS");
-        if (runningOnTravis != null && runningOnTravis.equals("true")) {
+        if (BuildConfig.TRAVIS != null && BuildConfig.TRAVIS.equals("true")) {
             // Skip all @Test methods in this class
-            Assume.assumeTrue("Skipping NavigationTest tests - this is expected behavior when running on Travis CI", false);
+            Assume.assumeTrue("Skipping NavigationTest tests - this is expected behavior when running on Travis CI (#1010)", false);
         }
     }
 

--- a/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
+++ b/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
@@ -16,6 +16,8 @@
 package org.onebusaway.android.nav.test;
 
 import org.apache.commons.io.IOUtils;
+import org.junit.Assume;
+import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.onebusaway.android.io.test.ObaTestCase;
@@ -53,6 +55,16 @@ public class NavigationTest extends ObaTestCase {
     private static final String TAG = "NavigationTest";
 
     private static final long SPEED_UP = 1000000L;
+
+    @BeforeClass
+    public static void beforeClass() {
+        // Only run this test suite if we're not running on Travis CI (it's too slow)
+        String runningOnTravis = System.getProperty("RUNNING_ON_TRAVIS");
+        if (runningOnTravis != null && runningOnTravis.equals("true")) {
+            // Skip all @Test methods in this class
+            Assume.assumeTrue("Skipping NavigationTest tests - this is expected behavior when running on Travis CI", false);
+        }
+    }
 
     /**
      * Started Stop: Mckinley Dr @ DOT Bldg

--- a/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
+++ b/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
@@ -17,10 +17,8 @@ package org.onebusaway.android.nav.test;
 
 import org.apache.commons.io.IOUtils;
 import org.junit.Assume;
-import org.junit.BeforeClass;
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.onebusaway.android.BuildConfig;
 import org.onebusaway.android.io.test.ObaTestCase;
 import org.onebusaway.android.mock.Resources;
 import org.onebusaway.android.nav.NavigationServiceProvider;
@@ -43,6 +41,7 @@ import androidx.test.runner.AndroidJUnit4;
 import static androidx.test.InstrumentationRegistry.getTargetContext;
 import static junit.framework.Assert.assertTrue;
 import static junit.framework.Assert.fail;
+import static org.onebusaway.android.util.TestUtils.isRunningOnTravis;
 
 /**
  * Loads previously recorded trips (GPS data) and replays them through the NavigationServiceProvider
@@ -57,15 +56,6 @@ public class NavigationTest extends ObaTestCase {
 
     private static final long SPEED_UP = 1000000L;
 
-    @BeforeClass
-    public static void beforeClass() {
-        // Only run this test suite if we're not running on Travis CI (it's too slow)
-        if (BuildConfig.TRAVIS != null && BuildConfig.TRAVIS.equals("true")) {
-            // Skip all @Test methods in this class
-            Assume.assumeTrue("Skipping NavigationTest tests - this is expected behavior when running on Travis CI (#1010)", false);
-        }
-    }
-
     /**
      * Started Stop: Mckinley Dr @ DOT Bldg
      * Destination Stop: University Area Transit Center
@@ -74,6 +64,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip1() throws IOException {
+        checkTravis();
         runSimulation("nav_trip1", 848, 978);
     }
 
@@ -85,6 +76,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip1C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip1c", 665, 929);
     }
 
@@ -96,7 +88,8 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip2() throws IOException {
-         runSimulation("nav_trip2", 0, 15);
+        checkTravis();
+        runSimulation("nav_trip2", 0, 15);
     }
 
     /**
@@ -118,6 +111,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip3() throws IOException {
+        checkTravis();
         runSimulation("nav_trip3", 95, 111);
     }
 
@@ -129,6 +123,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip3C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip3c", 87, 109);
     }
 
@@ -141,6 +136,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip4() throws IOException {
+        checkTravis();
         runSimulation("nav_trip4", 294, 329);
     }
 
@@ -153,6 +149,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip4C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip4c", 253, 329);
     }
 
@@ -164,6 +161,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip5() throws IOException {
+        checkTravis();
         runSimulation("nav_trip5", 372, 660);
     }
 
@@ -175,6 +173,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip5C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip5c", 284, 492);
     }
 
@@ -186,6 +185,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip6() throws IOException {
+        checkTravis();
         runSimulation("nav_trip6", 801, 837);
     }
 
@@ -197,6 +197,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip6C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip6c", 676, 704);
     }
 
@@ -208,6 +209,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip7() throws IOException {
+        checkTravis();
         runSimulation("nav_trip7", 183, 208);
     }
 
@@ -219,6 +221,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip7C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip7c", 172, 285);
     }
 
@@ -230,6 +233,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip8() throws IOException {
+        checkTravis();
         runSimulation("nav_trip8", 0, 18);
     }
 
@@ -241,6 +245,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip8C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip8c", 0, 20);
     }
 
@@ -252,6 +257,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip9() throws IOException {
+        checkTravis();
         runSimulation("nav_trip9", 1041, 1071);
     }
 
@@ -263,6 +269,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip9C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip9c", 570, 634);
     }
 
@@ -274,6 +281,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip10() throws IOException {
+        checkTravis();
         runSimulation("nav_trip10", 589, 605);
     }
 
@@ -285,6 +293,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip10C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip10c", 484, 509);
     }
 
@@ -296,6 +305,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip11() throws IOException {
+        checkTravis();
         runSimulation("nav_trip11", 331, 371);
     }
 
@@ -307,6 +317,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip11C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip11c", 119, 158);
     }
 
@@ -318,6 +329,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip12() throws IOException {
+        checkTravis();
         runSimulation("nav_trip12", 69, 199);
     }
 
@@ -329,6 +341,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip12C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip12c", 880, 901);
     }
 
@@ -340,6 +353,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip13() throws IOException {
+        checkTravis();
         runSimulation("nav_trip13", 2549, 2732);
     }
 
@@ -351,6 +365,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip13C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip13c", 230, 519);
     }
 
@@ -365,6 +380,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip14() throws IOException {
+        checkTravis();
         runSimulation("nav_trip14", 548, 571);
     }
 
@@ -376,6 +392,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip14C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip14c", 387, 429);
     }
 
@@ -387,6 +404,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip15() throws IOException {
+        checkTravis();
         runSimulation("nav_trip15", 225, 417);
     }
 
@@ -398,6 +416,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip15C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip15c", 217, 287);
     }
 
@@ -410,6 +429,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip16() throws IOException {
+        checkTravis();
         runSimulation("nav_trip16", 142, 192);
     }
 
@@ -421,6 +441,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip16C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip16c", 121, 168);
     }
 
@@ -432,6 +453,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip17() throws IOException {
+        checkTravis();
         runSimulation("nav_trip17", 100, 181);
     }
 
@@ -443,6 +465,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip18() throws IOException {
+        checkTravis();
         runSimulation("nav_trip18", 31, 51);
     }
 
@@ -454,6 +477,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip18C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip18c", 31, 58);
     }
 
@@ -465,6 +489,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip20() throws IOException {
+        checkTravis();
         runSimulation("nav_trip20", 133, 195);
     }
 
@@ -476,6 +501,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip20C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip20c", 53, 117);
     }
 
@@ -487,6 +513,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip21() throws IOException {
+        checkTravis();
         runSimulation("nav_trip21", 547, 686);
     }
 
@@ -498,6 +525,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip21C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip21c", 323, 402);
     }
 
@@ -509,6 +537,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip22() throws IOException {
+        checkTravis();
         runSimulation("nav_trip22", 1085, 1198);
     }
 
@@ -520,6 +549,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip22C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip22c", 1032, 1099);
     }
 
@@ -531,6 +561,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip23() throws IOException {
+        checkTravis();
         runSimulation("nav_trip23", 0, 24);
     }
 
@@ -542,6 +573,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip23C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip23c", 0, 19);
     }
 
@@ -553,6 +585,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip24() throws IOException {
+        checkTravis();
         runSimulation("nav_trip24", 115, 178);
     }
 
@@ -564,6 +597,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip24C() throws IOException {
+        checkTravis();
          runSimulation("nav_trip24c", 203, 262);
     }
 
@@ -575,7 +609,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip25() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip25", 209, 235);
     }
 
@@ -587,6 +621,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip25C() throws IOException {
+        checkTravis();
         runSimulation("nav_trip25c", 217, 247);
     }
 
@@ -598,7 +633,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip26() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip26", 367, 463);
     }
 
@@ -610,7 +645,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip27() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip27", 909, 944);
     }
 
@@ -622,7 +657,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip28() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip28", 400, 473);
     }
 
@@ -634,7 +669,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip29() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip29", 472, 594);
     }
 
@@ -646,7 +681,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip30() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip30", 1001, 1048);
     }
 
@@ -658,7 +693,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip31() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip31", 2508, 2654);
     }
 
@@ -671,7 +706,7 @@ public class NavigationTest extends ObaTestCase {
 
     @Test
     public void testTrip32() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip32", 1847, 1888);
     }
 
@@ -683,7 +718,7 @@ public class NavigationTest extends ObaTestCase {
      */
     @Test
     public void testTrip33() throws IOException {
-        // Read test CSV
+        checkTravis();
         runSimulation("nav_trip33", 335, 401);
     }
 
@@ -919,6 +954,16 @@ public class NavigationTest extends ObaTestCase {
 
             Boolean check2 = provider.getGetReady() && provider.getFinished();
             assertTrue(check2);
+        }
+    }
+
+    /**
+     * Checks to see if we're running on Travis, and if so, skips this test with Assume.assumeTrue().
+     * Note that we can't put this in a @BeforeClass method because otherwise we get "test run failed: 'Test run failed to complete. Expected 204 tests, received 149'"
+     */
+    public static void checkTravis() {
+        if (isRunningOnTravis()) {
+            Assume.assumeTrue("Skipping NavigationTest tests - this is expected behavior when running on Travis CI (#1010)", false);
         }
     }
 }

--- a/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
+++ b/onebusaway-android/src/androidTest/java/org/onebusaway/android/nav/test/NavigationTest.java
@@ -59,7 +59,7 @@ public class NavigationTest extends ObaTestCase {
     @BeforeClass
     public static void beforeClass() {
         // Only run this test suite if we're not running on Travis CI (it's too slow)
-        String runningOnTravis = System.getProperty("RUNNING_ON_TRAVIS");
+        String runningOnTravis = System.getenv("RUNNING_ON_TRAVIS");
         if (runningOnTravis != null && runningOnTravis.equals("true")) {
             // Skip all @Test methods in this class
             Assume.assumeTrue("Skipping NavigationTest tests - this is expected behavior when running on Travis CI", false);

--- a/onebusaway-android/src/androidTest/java/org/onebusaway/android/util/test/UIUtilTest.java
+++ b/onebusaway-android/src/androidTest/java/org/onebusaway/android/util/test/UIUtilTest.java
@@ -458,7 +458,6 @@ public class UIUtilTest extends ObaTestCase {
         assertEquals(options.get(3), "Edit this reminder");
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
-        assertEquals(options.get(6), "Join discussion");
         if (!isRunningOnTravis()) {
             assertEquals(options.get(6), "Join discussion");
             assertEquals(options.get(7), "About occupancy");

--- a/onebusaway-android/src/androidTest/java/org/onebusaway/android/util/test/UIUtilTest.java
+++ b/onebusaway-android/src/androidTest/java/org/onebusaway/android/util/test/UIUtilTest.java
@@ -17,12 +17,6 @@
 
 package org.onebusaway.android.util.test;
 
-import android.graphics.Color;
-import android.location.Location;
-import android.os.Bundle;
-import android.text.TextUtils;
-import android.widget.TextView;
-
 import org.junit.Test;
 import org.junit.runner.RunWith;
 import org.onebusaway.android.R;
@@ -46,6 +40,12 @@ import org.onebusaway.android.ui.ArrivalInfo;
 import org.onebusaway.android.util.ArrivalInfoUtils;
 import org.onebusaway.android.util.UIUtils;
 
+import android.graphics.Color;
+import android.location.Location;
+import android.os.Bundle;
+import android.text.TextUtils;
+import android.widget.TextView;
+
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.HashSet;
@@ -60,6 +60,7 @@ import static junit.framework.Assert.assertFalse;
 import static junit.framework.Assert.assertNotNull;
 import static junit.framework.Assert.assertNull;
 import static junit.framework.Assert.assertTrue;
+import static org.onebusaway.android.util.TestUtils.isRunningOnTravis;
 
 /**
  * Tests to evaluate utility methods related to the presentation of information to the user in the
@@ -214,7 +215,13 @@ public class UIUtilTest extends ObaTestCase {
         // a reminder and view the route schedule
         List<String> options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl, isReminderVisible, occupancy, occupancyState);
-        assertEquals(8, options.size());
+
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(7, options.size());
+        } else {
+            assertEquals(8, options.size());
+        }
         assertEquals(options.get(0), "Add star to route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
@@ -222,14 +229,22 @@ public class UIUtilTest extends ObaTestCase {
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Show route schedule");
         assertEquals(options.get(6), "Report arrival time problem");
-        assertEquals(options.get(7), "Join discussion");
+
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(7), "Join discussion");
+        }
 
         isReminderVisible = true;
 
         // Now we should see route schedules and *edit* the reminder
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl, isReminderVisible, occupancy, occupancyState);
-        assertEquals(8, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(7, options.size());
+        } else {
+            assertEquals(8, options.size());
+        }
         assertEquals(options.get(0), "Add star to route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
@@ -237,7 +252,9 @@ public class UIUtilTest extends ObaTestCase {
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Show route schedule");
         assertEquals(options.get(6), "Report arrival time problem");
-        assertEquals(options.get(7), "Join discussion");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(7), "Join discussion");
+        }
 
         // Get a PSTA response - PSTA test data doesn't include route schedule URLs
         ObaArrivalInfoResponse response2 =
@@ -266,28 +283,42 @@ public class UIUtilTest extends ObaTestCase {
         // user to set a reminder but NOT view the route schedule
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl2, isReminderVisible, occupancy, occupancyState);
-        assertEquals(7, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(6, options.size());
+        } else {
+            assertEquals(7, options.size());
+        }
         assertEquals(options.get(0), "Add star to route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
         assertEquals(options.get(3), "Set a reminder");
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
-        assertEquals(options.get(6), "Join discussion");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(6), "Join discussion");
+        }
 
         isReminderVisible = true;
 
         // Now we should see *edit* the reminder, and still no route schedule
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl2, isReminderVisible, occupancy, occupancyState);
-        assertEquals(7, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(6, options.size());
+        } else {
+            assertEquals(7, options.size());
+        }
         assertEquals(options.get(0), "Add star to route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
         assertEquals(options.get(3), "Edit this reminder");
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
-        assertEquals(options.get(6), "Join discussion");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(6), "Join discussion");
+        }
 
         // Now change route to favorite, and do all the above over again
         isRouteFavorite = true;
@@ -299,7 +330,12 @@ public class UIUtilTest extends ObaTestCase {
         // a reminder and view the route schedule
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl, isReminderVisible, occupancy, occupancyState);
-        assertEquals(8, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(7, options.size());
+        } else {
+            assertEquals(8, options.size());
+        }
         assertEquals(options.get(0), "Remove star from route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
@@ -307,14 +343,21 @@ public class UIUtilTest extends ObaTestCase {
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Show route schedule");
         assertEquals(options.get(6), "Report arrival time problem");
-        assertEquals(options.get(7), "Join discussion");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(7), "Join discussion");
+        }
 
         isReminderVisible = true;
 
         // Now we should see route schedules and *edit* the reminder
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl, isReminderVisible, occupancy, occupancyState);
-        assertEquals(8, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(7, options.size());
+        } else {
+            assertEquals(8, options.size());
+        }
         assertEquals(options.get(0), "Remove star from route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
@@ -322,7 +365,9 @@ public class UIUtilTest extends ObaTestCase {
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Show route schedule");
         assertEquals(options.get(6), "Report arrival time problem");
-        assertEquals(options.get(7), "Join discussion");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(7), "Join discussion");
+        }
 
         // PSTA
         isReminderVisible = false;  // We don't have views here, so just fake it
@@ -331,28 +376,42 @@ public class UIUtilTest extends ObaTestCase {
         // user to set a reminder but NOT view the route schedule
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl2, isReminderVisible, occupancy, occupancyState);
-        assertEquals(7, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(6, options.size());
+        } else {
+            assertEquals(7, options.size());
+        }
         assertEquals(options.get(0), "Remove star from route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
         assertEquals(options.get(3), "Set a reminder");
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
-        assertEquals(options.get(6), "Join discussion");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(6), "Join discussion");
+        }
 
         isReminderVisible = true;
 
         // Now we should see *edit* the reminder, and still no route schedule
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl2, isReminderVisible, occupancy, occupancyState);
-        assertEquals(7, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(6, options.size());
+        } else {
+            assertEquals(7, options.size());
+        }
         assertEquals(options.get(0), "Remove star from route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
         assertEquals(options.get(3), "Edit this reminder");
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
-        assertEquals(options.get(6), "Join discussion");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(6), "Join discussion");
+        }
 
         //
         // Test occupancy in the menu
@@ -363,22 +422,36 @@ public class UIUtilTest extends ObaTestCase {
         occupancyState = OccupancyState.HISTORICAL;
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl2, isReminderVisible, occupancy, occupancyState);
-        assertEquals(8, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(7, options.size());
+        } else {
+            assertEquals(8, options.size());
+        }
         assertEquals(options.get(0), "Remove star from route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
         assertEquals(options.get(3), "Edit this reminder");
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
-        assertEquals(options.get(6), "Join discussion");
-        assertEquals(options.get(7), "About historical occupancy");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(6), "Join discussion");
+            assertEquals(options.get(7), "About historical occupancy");
+        } else {
+            assertEquals(options.get(6), "About historical occupancy");
+        }
 
         // PREDICTED
         occupancy = Occupancy.EMPTY;
         occupancyState = OccupancyState.PREDICTED;
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl2, isReminderVisible, occupancy, occupancyState);
-        assertEquals(8, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(7, options.size());
+        } else {
+            assertEquals(8, options.size());
+        }
         assertEquals(options.get(0), "Remove star from route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
@@ -386,22 +459,36 @@ public class UIUtilTest extends ObaTestCase {
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
         assertEquals(options.get(6), "Join discussion");
-        assertEquals(options.get(7), "About occupancy");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(6), "Join discussion");
+            assertEquals(options.get(7), "About occupancy");
+        } else {
+            assertEquals(options.get(6), "About occupancy");
+        }
 
         // REALTIME (should be same as PREDICTED)
         occupancy = Occupancy.EMPTY;
         occupancyState = OccupancyState.REALTIME;
         options = UIUtils
                 .buildTripOptions(getTargetContext(), isRouteFavorite, hasUrl2, isReminderVisible, occupancy, occupancyState);
-        assertEquals(8, options.size());
+        if (isRunningOnTravis()) {
+            // We don't have the Embedded Social config set up on Travis, so it doesn't include that menu item.
+            assertEquals(7, options.size());
+        } else {
+            assertEquals(8, options.size());
+        }
         assertEquals(options.get(0), "Remove star from route");
         assertEquals(options.get(1), "Show route on map");
         assertEquals(options.get(2), "Show trip status");
         assertEquals(options.get(3), "Edit this reminder");
         assertEquals(options.get(4), "Show only this route");
         assertEquals(options.get(5), "Report arrival time problem");
-        assertEquals(options.get(6), "Join discussion");
-        assertEquals(options.get(7), "About occupancy");
+        if (!isRunningOnTravis()) {
+            assertEquals(options.get(6), "Join discussion");
+            assertEquals(options.get(7), "About occupancy");
+        } else {
+            assertEquals(options.get(6), "About occupancy");
+        }
     }
 
     @Test

--- a/onebusaway-android/src/main/java/org/onebusaway/android/util/TestUtils.java
+++ b/onebusaway-android/src/main/java/org/onebusaway/android/util/TestUtils.java
@@ -82,4 +82,13 @@ public class TestUtils {
     public static boolean isRunningOnEmulator() {
         return Build.FINGERPRINT.contains("generic");
     }
+
+    /**
+     * Returns true if the test is running on Travis CI, and false if it is not
+     *
+     * @return true if the test is running on Travis CI, and false if it is not
+     */
+    public static boolean isRunningOnTravis() {
+        return BuildConfig.TRAVIS != null && BuildConfig.TRAVIS.equals("true");
+    }
 }

